app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionPanelView.kt

```diff
@@ -0,0 +1,217 @@
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.getHostFromUrl
+import org.mozilla.fenix.trackingprotection.TrackingProtectionCategory.CROSS_SITE_TRACKING_COOKIES
+import org.mozilla.fenix.trackingprotection.TrackingProtectionCategory.CRYPTOMINERS
+import org.mozilla.fenix.trackingprotection.TrackingProtectionCategory.FINGERPRINTERS
+import org.mozilla.fenix.trackingprotection.TrackingProtectionCategory.SOCIAL_MEDIA_TRACKERS
+import org.mozilla.fenix.trackingprotection.TrackingProtectionCategory.TRACKING_CONTENT
+
+/**
+ * Interface for the TrackingProtectionPanelViewInteractor. This interface is implemented by objects that want
+ * to respond to user interaction on the TrackingProtectionPanelView
+ */
+interface TrackingProtectionPanelViewInteractor {
+    /**
+     * Called whenever the settings option is tapped
+     */
+    fun selectTrackingProtectionSettings()
+
+    /**
+     * Called whenever the tracking protection toggle for this site is toggled
+     * @param isEnabled new status of session tracking protection
+     */
+    fun trackingProtectionToggled(isEnabled: Boolean)
+
+    /**
+     * Called whenever back is pressed
+     */
+    fun onBackPressed()
+
+    /**
+     * Called whenever an active tracking protection category is tapped
+     * @param category The Tracking Protection Category to view details about
+     */
+    fun openDetails(category: TrackingProtectionCategory)
+}
+
+/**
+ * View that contains and configures the Tracking Protection Panel
+ */
+class TrackingProtectionPanelView(
+    override val containerView: ViewGroup,
+    val interactor: TrackingProtectionPanelInteractor
+) : LayoutContainer {
+    val view: ConstraintLayout = LayoutInflater.from(containerView.context)
+        .inflate(R.layout.component_tracking_protection_panel, containerView, true)
+        .findViewById(R.id.panel_wrapper)
+
+    private val context get() = view.context
+
+    var mode: TrackingProtectionState.Mode = TrackingProtectionState.Mode.Normal
+        private set
+
+    var trackers: List<Tracker> = listOf()
+        private set
+
+    var bucketedTrackers: HashMap<TrackingProtectionCategory, List<String>> = HashMap()
+
+    fun update(state: TrackingProtectionState) {
+        if (state.mode != mode) {
+            mode = state.mode
+        }
+
+        if (state.listTrackers != trackers) {
+            trackers = state.listTrackers
+            bucketedTrackers = getHashMapOfTrackersForCategory(state.listTrackers)
+        }
+
+        when (val mode = state.mode) {
+            is TrackingProtectionState.Mode.Normal -> setUIForNormalMode(state)
+            is TrackingProtectionState.Mode.Details -> setUIForDetailsMode(mode.selectedCategory)
+        }
+    }
+
+    private fun setUIForNormalMode(state: TrackingProtectionState) {
+        details_mode.visibility = View.GONE
+        normal_mode.visibility = View.VISIBLE
+
+        // TODO I don't think we have this list yet of "could be blocking" trackers
```

Comment:
@Amejia481 is this correct? We can't do the "not blocking" list yet?

Reply:
Yes, for the not blocking list we have [an GV bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1567611)